nionswift_plugin/nion_hyperspy/HyperSpyUI.py

```diff
@@ -20,7 +20,8 @@
 import nion.hyperspy
 
 signal = nion.hyperspy.xdata_to_signal(src.display_xdata)
-fit_px = int(fit_region.interval[0] * src.display_xdata.data_shape[0]), int(fit_region.interval[1] * src.display_xdata.data_shape[0])
+calibration = src.display_xdata.dimensional_calibrations[0]
+fit_px = calibration.convert_to_calibrated_value(int(fit_region.interval[0] * src.display_xdata.data_shape[0])), calibration.convert_to_calibrated_value(int(fit_region.interval[1] * src.display_xdata.data_shape[0]))
```

Comment:
I'm not sure if the original code is wrong. The `remove_background` function isn't fully documented, but according to the `signal_range` parameter in [the documentation](http://hyperspy.org/hyperspy-doc/current/api/hyperspy._signals.html?highlight=remove_background#hyperspy._signals.signal1d.Signal1D.align_zero_loss_peak) for `align_zero_loss_peak`,  passing integers indicates you are specifying indexes rather than spectrum values. Presumably indexes refer to indexes into the raw data and spectrum values refer to calibrated values. Are you seeing incorrect behavior?

Reply:
Yes I see incorrect behavior, which is that the fit range that you set in Swift is definitely not the fit range it is using internally. So I started looking for the bug. I agree that the hyperspy documentation is not great so I had a look at the [source code](https://github.com/hyperspy/hyperspy/blob/fabb93f0010f697a671a6df112392b53d6c84d59/hyperspy/_signals/signal1d.py#L1032) of `remove_background`. Internally it uses `Model1D` and a background estimator from `components1d`. For example in the [source code](https://github.com/hyperspy/hyperspy/blob/fabb93f0010f697a671a6df112392b53d6c84d59/hyperspy/_components/power_law.py#L24) of `PowerLaw` (which is what is used in Swift atm) you can see that in `estimate_parameters` it converts the spectral range input to indices (see [line 108](https://github.com/hyperspy/hyperspy/blob/fabb93f0010f697a671a6df112392b53d6c84d59/hyperspy/_components/power_law.py#L108)).
This told me that `remove_background` wants calibrated units as an input. And indeed after I changed it the background removal in Swift worked as expected.

Reply:
Ok - I guess we should push a documentation fix upstream to HyperSpy too...

So there are two uses of `remove_background` in that file. Can you fix the other one too?

Also, the name `fit_px` is no longer accurate. How about `fit_eV` or `fit_calibrated` for a variable name?

Reply:
How about `fit_range`? Or do you prefer if the name implies that it contains calibrated units?

Reply:
`fit_range` sounds good to me. It was the `_px` suffix that was wrong!

Reply:
Allright then I'll fix the second occurance of `remove_background` too and send you another pull request (also with the updated variable name)